app/tut/ga/creatlach.md

```diff
@@ -0,0 +1,967 @@
+}
+
+mise := Duine("Eoin")
+```
+
+Bain triail as an gcód anois. Ba chóir go feicfidh tú "Cruthaíodh duine: Eoin" ar an gconsól.
+
+{{{
+creatlach Duine {
+    gníomh nua(ainm) {
+        scríobh("Cruthaíodh duine:", ainm)
+    }
+}
+
+mise := Duine("Eoin")
+}}}
+
+### Cruthaitheoir Ciorcail
+
+Toisc gur chonaiceamar cruthaitheoirí, is féidir linn an gníomh `déan_ciorcal` a scriosadh. In
+ionad an gníomh sin bainfimid úsáid as cruthaitheoir chun `x`, `y`, `ga` agus `dath` an ciorcail a
+stóráil san ásc atá á chruthú againn.
+
+Seo é ár gcruthaitheoir:
+
+```{.setanta .numberLines}
+gníomh nua(x, y, ga, dath) {
+    x@seo = x
+    y@seo = y
+    ga@seo = ga
+    dath@seo = dath
+}
+```
+
+Cuirimis an cruthaitheoir sin sa chreatlach `Ciorcal`, agus scriosaimis an gníomh `déan_ciorcal` go
+hiomlán. Anois nuair ba mhaith linn ásc `Ciorcal` a chruthú is féidir linn an gníomh `Ciorcal(x, y,
+ga, dath)`{.setanta} a úsáid go díreach. Mar shampla:
+
+```{.setanta .numberLines}
+c := Ciorcal(100, 100, 50, "dearg")
+```
+
+Anois seo é ár ríomhchlár:
+
+{{{s
+>-- Creatlach folamh.
+creatlach Ciorcal {
+    gníomh nua(x, y, ga, dath) {
+        x@seo = x
+        y@seo = y
+        ga@seo = ga
+        dath@seo = dath
+    }
+
+    >-- Tarraing an ciorcal.
+    gníomh tarraing() {
+        >-- Athraigh dath an phinn.
+        dath@stáitse(dath@seo)
+
+        >-- Tarraing an ciorcal leis an x comhordanáid, y comhordanáid agus an ga.
+        ciorcal@stáitse(x@seo, y@seo, ga@seo)
+    }
+
+    >-- Ríomh an t-achar. (achar = pi * ga^2).
+    gníomh achar() {
+        toradh pi@mata * ga@seo * ga@seo
+    }
+
+    >-- Ríomh an t-imlíne. (imlíne = 2 * pi * ga).
+    gníomh imlíne() {
+        toradh 2 * pi@mata * ga@seo
+    }
+}
+
+c := Ciorcal(100, 100, 50, "dearg")
+tarraing@c()
+scríobh("Achar: ", achar@c())
+scríobh("Imlíne: ", imlíne@c())
+}}}
+
+Chonaiceamar anseo gur féidir linn oibiachtaí a úsáid chun cód níos deise a scríobh, ach tá siad i
+bhfad [[níos cumhachtaí|more powerful]] ná sin fós!
+
+# [[Oidhreacht|Inheritance]]
+
+I *Setanta*, is féidir le creatlach amháin [[iompraíocht|behaviour]] a fháil
+[[le hoidhreacht|through inheritance]] ó gcreatlach eile. Cad is brí le sin? Is féidir linn
+creatlach [["tuismitheoir"|parent]] a chruthú, agus roinnt creatlacha [["páiste"|child]]; Ansin
+beidh gach gníomh creatlaí atá ag an **[[máthair-creatlach|parent-outline]]** ag gach
+**[[mac-creatlach|child-outline]]** freisin. (Is féidir linn [["for-creatlach"|super-outline]] agus
+"[[fo-creatlach|sub-outline]]" a thabhairt ar na gcreatlacha sin freisin.)
+
+Tá sé a lán níos soiléire le sampla:
+
+## Sampla
+
+Féach anois ar an gcreatlach seo:
+
+```{.setanta .numberLines}
+creatlach Ainmhí {
+    gníomh bog() {
+        scríobh("Táim ag bogadh")
+    }
+}
+```
+
+Níl ach gníomh amháin aige (`bog`) agus nuair a ghlaoitear air, scríobhann sé "Táim ag bogadh" ar an
+gconsól.
+
+Anois cruthaímis creatlach nua `Madra` atá mac-creatlach do `Ainmhí`. Úsáidimid an
```

Comment:
ina mhac-creatlach ag ainmhí

(b'fhéidir go mbeadh for-/fo- tríd síos níos fearr!)